sqlmesh/core/model/definition.py

```diff
@@ -1598,8 +1598,11 @@ def create_seed_model(
             from the macro registry.
     """
     seed_path = Path(seed_kind.path)
-    if not seed_path.is_absolute():
-        seed_path = path / seed_path if path.is_dir() else path.parents[0] / seed_path
+    base, *subdir = seed_path.parts
+    if base == "~":
```

Comment:
Any chance we can use [expanduser()](https://docs.python.org/3/library/pathlib.html#pathlib.Path.expanduser) instead?

Reply:
Ah, sorry, this is not correct. I didn't realize we overload `~` to mean the project path. Not sure I like such implicit overloading. I feel like `~` is so universally used that substituting its meaning like can cause quite a bit of confusion.

Reply:
Sure, I commented the same thought. What symbol should we prefer. We need a way to indicate the context root. 

How about something longer in the flavor of bash substitution `$root/seeds/whatever.csv`